### DISCUSSION TITLE
Have trouble to understand alignment_mode arguments

### DISCUSSION THREAD

**Discussion by innocent-charles** [explosion/spacy-llm#378] [Q&A]
``` The alignment_mode argument is used to match entities as returned by the LLM to the tokens from the original Doc - specifically it’s used as argument in the call to [doc.char_span()](https://spacy.io/api/doc#char_span). The "strict" mode will only keep spans that strictly adhere to the given token boundaries. "contract" will only keep those tokens that are fully within the given range, e.g. reducing "New Y" to "New". Finally, "expand" will expand the span to the next token boundaries, e.g. expanding "New Y" out to "New York".```

I have document as "Innocent Charles is a Tanzanian" and i wanted to extract the country from it. 

LLMs give me country : Tanzania when i check the logs but, i dont get anything in the docs.ents output. 

Why and How that ?

**Comment by innocent-charles** [comment]
even for the "Innocent Charles is a Tanzanian, born on 22th, October 1999" . when i want to extract the day innocent was born, the LLMs give 22 when I check the logs, but the docs.ents do not provide anything.

**Comment by rmitsch** [comment]
Hi @innocent-charles! Please 
- post your config and your code and
- run this with `save_io` on and post your output.

I'm not sure whether `alignment_mode` is relevant to this at all.

**Comment by innocent-charles** [comment]
Okay, but how to run it with save_io ?

**Reply by rmitsch** [reply]
You can pass `save_io` in to the LLM component in the config. See the [docs](https://spacy.io/api/large-language-models#init). I. e. in your config:
```ini
[components.llm]
factory = "llm"
save_io = True
```

**Reply by innocent-charles** [reply]
Okay ! Thanks for help @rmitsch

**Reply by innocent-charles** [reply]
sorry, how to access the saved LLM(prompts and responses) i just dont see the file or something..

**Reply by rmitsch** [reply]
Ah, there is a mistake in the docs here:
> Whether to save LLM I/O (prompts and responses) in the `Doc._.llm_io` custom attribute. 

`Doc._.llm_io` should actually be `Doc.user_data["llm_io"]`. We'll fix the docs.
